src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -266,26 +272,33 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
         /// <param name="messageProvider">Used to produce diagnostics.</param>
         /// <param name="diagnostics">Any produced diagnostics are added to this bag.</param>
         /// <returns>
-        /// A list of diagnostic options, where the dictionary index corresponds
+        /// A list of diagnostic and analyzer options, where the dictionary index corresponds
         /// to the options for the source path at the same index. If there are
-        /// no diagnostic options for the given path, the entry is null. Otherwise,
-        /// the options are a map from diagnostic ID to diagnostic severity, and
+        /// no options for the given path, the entry is null. Otherwise, the diagnostic
+        /// options are a map from diagnostic ID to diagnostic severity, and
         /// correspond to options provided for <see cref="SyntaxTree.DiagnosticOptions"/>.
+        /// The analyzerOptions are an OptionSet consisting of all properties that were not
+        /// diagnostic options.
         /// </returns>
-        internal static ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> GetAnalyzerConfigOptions(
-            IReadOnlyList<string> sourcePaths,
-            ArrayBuilder<EditorConfig> analyzerConfigs,
-            CommonMessageProvider messageProvider,
-            DiagnosticBag diagnostics)
+        internal static (ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> treeOptions,
+                         ImmutableArray<ImmutableDictionary<string, string>> analyzerOptions)
+            GetAnalyzerConfigOptions(IReadOnlyList<string> sourcePaths,
+                                     IReadOnlyList<EditorConfig> analyzerConfigs,
+                                     CommonMessageProvider messageProvider,
+                                     DiagnosticBag diagnostics)
         {
-            // Sort editorconfig paths from shortest to longest.
             // Since paths are compared as ordinal string comparisons, the least nested
             // file will always be the first in sort order
-            analyzerConfigs.Sort(Comparer<EditorConfig>.Create(
-                (e1, e2) => e1.NormalizedDirectory.Length.CompareTo(e2.NormalizedDirectory.Length)));
+            if (!analyzerConfigs.IsSorted(Comparer<EditorConfig>.Create(
+                (e1, e2) => e1.NormalizedDirectory.Length.CompareTo(e2.NormalizedDirectory.Length))))
```

Comment:
indentaiton is wonky here.  it makes it look like the lambda is a sibling of hte !... expression (when really it's a child of it).